File: RepoTransBench/target_projects/Python/C++/antiboredom_audiogrep/public_tests/test_audiogrep_public.cpp
#include <gtest/gtest.h>
#include <fstream>
#include <string>
#include <vector>
#include <set>
#include <algorithm>

// Simulate: std::set<std::string> get_files(const std::string&, const std::vector<std::string>&);
namespace audiogrep {
    std::set<std::string> get_files(const std::string& directory, const std::vector<std::string>& exts);
}

TEST(PublicAudiogrepTest, GetFilesPublic)
{
    // Simulate creation of several files in a temp dir
    std::string tmp = "/tmp/audiogrep_pub_" + std::to_string(rand());
    system(("mkdir -p " + tmp).c_str());
    std::string a = tmp + "/file1.aac";
    std::string b = tmp + "/file2.m4a";
    std::string c = tmp + "/file3.txt";
    std::ofstream(a) << "test abc";
    std::ofstream(b) << "test abc";
    std::ofstream(c) << "test abc";
    std::set<std::string> fs = audiogrep::get_files(tmp, {".aac", ".m4a"});
    std::set<std::string> expected = {a, b};
    EXPECT_EQ(fs, expected);
    system(("rm -rf " + tmp).c_str());
}

File: RepoTransBench/target_projects/Python/Rust/antiboredom_audiogrep/public_tests/test_audiogrep_public.rs
use antiboredom_audiogrep::*;
use tempfile::tempdir;
use std::fs;
use std::io::Write;

#[test]
fn test_get_files_public() {
    let tmp = tempdir().unwrap();
    let a = tmp.path().join("file1.aac");
    let b = tmp.path().join("file2.m4a");
    let c = tmp.path().join("file3.txt");
    for f in [&a, &b, &c] {
        let mut file = fs::File::create(f).unwrap();
        write!(file, "test abc").unwrap();
    }
    let fs = get_files(tmp.path(), &[".aac", ".m4a"]);
    let expected = vec![a.to_string_lossy().to_string(), b.to_string_lossy().to_string()];
    assert_eq!(std::collections::HashSet::<_>::from_iter(fs.iter()), std::collections::HashSet::from_iter(expected.iter()));
}